.github/workflows/check-static.yml

```diff
@@ -0,0 +1,49 @@
+# Copyright (C) 2020  Matthew Glazar
+# See end of file for extended copyright information.
+
+name: static analysis
+on:
+  push:
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  analysis:
+    name: clang-tidy
+    runs-on: ubuntu-latest
+    container: ghcr.io/quick-lint/quick-lint-js-github-builder:v1
+    env:
+      C: clang-9
+      CFLAGS: "-stdlib=libc++"
+      CXX: clang++-9
+      CXXFLAGS: "-std=gnu++17 -stdlib=libstdc++ -O2"
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: clang-tidy install
+        run: |
+          apt-get update
+          apt-get install -y clang-tidy-9
+      - name: CMake Build with clang-tidy enabled
+        run: cmake -DCMAKE_C_COMPILER=clang-9 -DCMAKE_CXX_COMPILER=clang++-9 -DCMAKE_C_FLAGS="${CFLAGS}" -DCMAKE_CXX_FLAGS="${CXXFLAGS}" -DBUILD_TESTING=OFF -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_CLANG_TIDY="clang-tidy-9;-checks=clang-analyzer-*;--export-fixes=clang-tidy-output.yml" -S . -B build
+      - name: build with clang-tidy static analysis
+        run: cmake --build build -j9 1>/dev/null
+      - name: clang-tidy output
+        run: cat ./build/src/clang-tidy-output.yml
```

Comment:
In order for someone to see the report, they need to look for it, right? I doubt anyone is going to look for the report by hand. Do you plan on making this job post a comment on the PR or send an email? Or fail the build if there are any issues found? If not, then unless we develop a habit of looking at the reports, this job is a waste of CPU time.

Reply:
per suggestion. I have added a fail condition on the latest commit. That should help making the action more noticeable.